src/sql/workbench/contrib/query/test/browser/queryInputFactory.test.ts

```diff
@@ -205,16 +416,25 @@ class MockConnectionManagementService extends TestConnectionManagementService {
 
 	public numberConnects = 0;
 
+	public connectionProfiles = [];
+
 	public override isProfileConnected(connectionProfile: IConnectionProfile): boolean {
 		return true;
 	}
 
 	public override connect(connection: IConnectionProfile, uri: string, options?: IConnectionCompletionOptions, callbacks?: IConnectionCallbacks): Promise<IConnectionResult> {
 		this.numberConnects++;
+		this.connectionProfiles.push(uri);
 		return Promise.resolve(undefined);
 	}
 
 	public override getConnectionProfile(fileUri: string): IConnectionProfile {
-		return <IConnectionProfile>{}; // Not actually used so fine to cast
+		let element = this.connectionProfiles.find(element => fileUri === element);
+		if (element) {
+			return <IConnectionProfile>{ azureResourceId: element };
```

Comment:
Yeah, not really sure what this is supposed to be doing. This isn't what azureResourceId is supposed to have. 

Why aren't you caching the actual profiles passed in during connect and then returning those instead?

Reply:
Doing that right now